product_variant_configurator/models/product_configurator.py

```diff
@@ -0,0 +1,254 @@
+            self.product_attribute_ids = attribute_lines
+
+        # Restrict product possible values to current selection
+        domain = [('product_tmpl_id', '=', self.product_tmpl_id.id)]
+        return {'domain': {'product_id': domain}}
+
+    @api.onchange('product_attribute_ids')
+    def onchange_product_attribute_ids(self):
+        if not self.product_attribute_ids:
+            return {}
+        product_obj = self.env['product.product']
+        domain, cont = product_obj._build_attributes_domain(
+            self.product_tmpl_id, self.product_attribute_ids)
+        self.product_id = False
+        if cont:
+            products = product_obj.search(domain)
+            # Filter the product with the exact number of attributes values
+            for product in products:
+                if len(product.attribute_value_ids) == cont:
+                    self.product_id = product.id
+                    break
+        if not self.product_id:
+            product_tmpl = self.product_tmpl_id
+            values = self.product_attribute_ids.mapped('value_id')
+            if 'partner_id' in self._fields:
+                # If our model has a partner_id field, language is got from it
+                obj = self.env['product.attribute.value'].with_context(
+                    lang=self.partner_id.lang)
+                values = obj.browse(
+                    self.product_attribute_ids.mapped('value_id').ids)
+                obj = self.env['product.template'].with_context(
+                    lang=self.partner_id.lang)
+                product_tmpl = obj.browse(self.product_tmpl_id.id)
+            self.name = self._get_product_description(
+                product_tmpl, False, values)
+        return {'domain': {'product_id': domain}}
+
+    @api.multi
+    def onchange_product_id_product_configurator(self):
+        # First, empty current list
+
+        self.ensure_one()
+        if self.product_id:
+            self.product_attribute_ids = [
+                (2, x.id) for x in self.product_attribute_ids]
+            attribute_list = (
+                self.product_id._get_product_attributes_values_dict())
+            for val in attribute_list:
+                val['product_tmpl_id'] = self.product_id.product_tmpl_id
+                val['owner_model'] = self._name
+            product = self.product_id
+            if 'partner_id' in self._fields:
+                # If our model has a partner_id field, language is got from it
+                product = self.env['product.product'].with_context(
+                    lang=self.partner_id.lang).browse(self.product_id.id)
+            self.product_attribute_ids = [(0, 0, x) for x in attribute_list]
+            self.product_tmpl_id = product.product_tmpl_id.id
+            self.name = self._get_product_description(
+                product.product_tmpl_id, product, product.attribute_value_ids)
+
+    @api.multi
+    def onchange_product_id_product_configurator_old_api(
```

Comment:
As this is a "framework", I would let it until v10 where old API is removed, although it has not been used in sale nor purchase modules.